src/database/type-transforms.tsx

```diff
@@ -15,8 +15,16 @@ import {
 import { referenceFacilitiesProp } from ".";
 import { FacilityDocUpdate } from "./types";
 
-const timestampToDate = (timestamp: firebase.firestore.Timestamp): Date => {
-  return timestamp.toDate();
+const timestampToDate = (
+  timestamp: firebase.firestore.Timestamp | null,
+): Date => {
+  if (timestamp) {
+    return timestamp.toDate();
+  } else {
+    // when Firestore writes are pending, server timestamps may be null;
+    // we can substitute the local now in the meantime
+    return new Date();
```

Comment:
I see that in the call to this below for `scenario.updatedAt`, it used to substitute `firebase.firestore.Timestamp.now()` which _looks_ like a different type than just `new Date()`. Am I mistaken? If not, why is this type change needed?

Reply:
rather than requiring all callers to catch nulls on their own and replace them with the local now in `Timestamp` form so that we can immediately convert it to a `Date`, we are now catching the nulls inside this function. When that happens there is nothing to convert, so we can just return a `Date` directly